src/cli/Cli.ts

```diff
@@ -35,16 +37,21 @@ export class Cli {
   public currentCommandName: string | undefined;
   private optionsFromArgs: { options: minimist.ParsedArgs } | undefined;
   private commandsFolder: string = '';
+  private showHelpOnFailure: boolean = false;
   private static instance: Cli;
+  private static conf: any;
 
   private constructor() {
+    Cli.conf = new Configstore(configstore.name);
   }
 
   public static getInstance(): Cli {
     if (!Cli.instance) {
       Cli.instance = new Cli();
     }
 
+    Cli.instance.showHelpOnFailure = Cli.conf.get(configstore.showHelpOnFailure) || false;
```

Comment:
Since we need this property only when an error occurred, I'd avoid setting it here and load it on demand where we handle errors and show help (or not). It will allow us to avoid overhead.